chat-sdk/src/main/java/com/amazon/connect/chat/sdk/repository/ChatService.kt

```diff
@@ -472,6 +485,32 @@ class ChatServiceImpl @Inject constructor(
         }
     }
 
+    override suspend fun resendFailedMessage(messageId: String): Result<Boolean> {
+        val oldMessage = transcriptDict[messageId] as? Message
+
+        // cannot retry if old message didn't exist or fail to be sent
+        if (oldMessage == null || !arrayOf(MessageStatus.Failed, MessageStatus.Custom, MessageStatus.Unknown).contains(oldMessage.metadata?.status)) {
+            return Result.failure(Exception("Unable to find the failed message"))
+        }
+
+        // remove failed message from transcript & transcript dict
+        internalTranscript.removeAll { it.id == messageId }
+        transcriptDict.remove(messageId);
+        // Send out updated transcript with old message removed
+        coroutineScope.launch {
+            _transcriptListPublisher.emit(internalTranscript)
+        }
+
+        // as the next step, attempt to resend the message based on its type
+        val attachmentUrl = tempMessageIdToFileUrl[messageId];
+        // if old message is an attachment
+        if (attachmentUrl != null) {
+            return sendAttachment(attachmentUrl)
+        } else {
+            return sendMessage(ContentType.fromType(oldMessage.contentType)!!, oldMessage.text)
```

Comment:
as mentioned by Michael, here also we could avoid using `!!` 
if the contentType is null it will be NPE.

Reply:
addressed in new rev